src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -6,31 +6,32 @@ import styles from './HeatmapVis.module.css';
 import ColorBar from './ColorBar';
 import Mesh from './Mesh';
 import TooltipMesh from '../shared/TooltipMesh';
-import { useHeatmapConfig } from './config';
 import PanZoomMesh from '../shared/PanZoomMesh';
 import VisCanvas from '../shared/VisCanvas';
 import { getDims } from './utils';
 import { findDomain } from '../shared/utils';
 import { useMemoColorScaleDomain } from './hooks';
+import type { HeatmapConfig } from './config';
+import type { Domain } from '../shared/models';
 
 interface Props {
   dataArray: ndarray<number>;
+  config: HeatmapConfig;
+  onDataDomainChange: (dataDomain: Domain | undefined) => void;
```

Comment:
I don't like this, because it makes users responsible for resetting the `requestedDomain` in `onDataDomainChange`, which they could do incorrectly. That and the API is not self-documenting.